### PR TITLE
jubn codeReview1

### PR DESCRIPTION
**Pull request by SoftwareTesting111345**
Analysis of the program.
This program is used to remind people of the books they lend to someone.

Was the program available in UC Github on time?
Yes

Is the program documented/commented well enough for you to understand?
Yes

Does the program compile?
Yes

Rationale behind your changes.
Add unit test for the project. URL: https://developer.android.com/training/testing/fundamentals
Add @SerializedName for the project URL:https://howtodoinjava.com/gson/gson-serializedname/
Add application class for the project. URL:https://docs.sonarqube.org/display/SONARQUBE45/Developers%27+Seven+Deadly+Sins

Links to three specific commits that you made to your own group's GitHub repository before the end of sprint deadline.  
If you do not post links to three commits that you made to your group's GitHub repo, your code review grade will be subject to a 20% penalty.
https://github.com/trey368s/Mobile-Device-Programming/commit/8c6ee8a4d3dc839b6cf9ea147f4c982df4f7d223
https://github.com/trey368s/Mobile-Device-Programming/commit/3dd3352bfe540419cd53d802bef1670e430b678b
https://github.com/trey368s/Mobile-Device-Programming/commit/a17866b2c93c21515d0bf15ae99bcf560c2f5013
https://github.com/trey368s/Mobile-Device-Programming/commit/693935abb52a1fb52909e64e81035ef1e36065cb

I learned how to add Koin in the project and write unit test.

### CODE REVIEW

File: app/src/test/java/com/example/mylittlelibrary/BookUnitTest.kt [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+import com.example.mylittlelibrary.data.Book
+import org.junit.Assert
+import org.junit.Test
+import java.util.*
+
+class BookUnitTest {
+    @Test
+    fun `given a book dto when name is MyLittleLibrary and lendTo is MyLittleLibraryI then code is MyLittleLibrary and name is MyLittleLibraryI`() {
+        var country = Book("MyLittleLibrary", "MyLittleLibraryI", Date())
```
Comment (SoftwareTesting111345) [inline app/src/test/java/com/example/mylittlelibrary/BookUnitTest.kt:11]: Add unit test for the project. URL: https://developer.android.com/training/testing/fundamentals

File: app/src/main/java/com/example/mylittlelibrary/data/Book.kt [hunk 2]
```diff
@@ -1,5 +1,12 @@
 package com.example.mylittlelibrary.data
-
+import com.google.gson.annotations.SerializedName
 import java.util.*
 
-data class Book(var name : String, var lendTo: String, var date: Date)
+data class Book(@SerializedName("name") var name : String, var lendTo: String, var date: Date)
```
Comment (SoftwareTesting111345) [inline app/src/main/java/com/example/mylittlelibrary/data/Book.kt:5]: Add @SerializedName for the project URL:https://howtodoinjava.com/gson/gson-serializedname/

File: app/src/main/java/com/example/mylittlelibrary/MyLittleLibraryApplication.kt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.example.mylittlelibrary
+
+import android.app.Application
+
```
Comment (SoftwareTesting111345) [inline app/src/main/java/com/example/mylittlelibrary/MyLittleLibraryApplication.kt:4]: Add application class for the project. URL:https://docs.sonarqube.org/display/SONARQUBE45/Developers%27+Seven+Deadly+Sins

File: app/build.gradle [hunk 4]
```diff
@@ -47,4 +47,6 @@ dependencies {
     testImplementation 'junit:junit:4.+'
     androidTestImplementation 'androidx.test.ext:junit:1.1.3'
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
+    implementation "io.insert-koin:koin-android:3.1.4"
```
Comment (SoftwareTesting111345) [inline app/build.gradle:50]: Add dependency for Koin URL:https://docs.sonarqube.org/display/SONARQUBE45/Developers%27+Seven+Deadly+Sins

File: app/src/main/java/com/example/mylittlelibrary/data/Book.kt [hunk 5]
```diff
@@ -1,5 +1,12 @@
...
+{
+    val bookString= name+" "+lendTo
+    override fun toString(): String {
```
Comment (SoftwareTesting111345) [inline app/src/main/java/com/example/mylittlelibrary/data/Book.kt:8]: Add a toString method to the class. URL:https://docs.sonarqube.org/display/SONARQUBE45/Developers%27+Seven+Deadly+Sins

File: app/src/main/java/com/example/mylittlelibrary/MyLittleLibraryApplication.kt (hunk 3 above)
Comment (discospiff) [reply app/src/main/java/com/example/mylittlelibrary/MyLittleLibraryApplication.kt:4]: But... it's not doing anything yet.  It is overriding a function, and then invoking that function.

File: app/src/main/java/com/example/mylittlelibrary/data/Book.kt (hunk 2 above)
Comment (discospiff) [reply app/src/main/java/com/example/mylittlelibrary/data/Book.kt:5]: Ahhhhhhh... it's not adding value if it's the same name (name) and capitalization as the attribute it is describing (name).

File: app/src/main/java/com/example/mylittlelibrary/data/Book.kt (hunk 5 above)
Comment (discospiff) [reply app/src/main/java/com/example/mylittlelibrary/data/Book.kt:8]: Use a Kotlin shortcut: return "$name %lendTo"

**Comment by discospiff** [comment]
From this review...
Additional unit test code coverage is always a good idea, and reduces technical debt.
Other changes... feel neutral to technical debt.  See my notes in Files Changed section.

**Comment by andrewglanzzz** [comment]
Closing this branch but will be taking recommendations from it. Not merging.

**Pull request closed without merging**